function-server/src/main/java/io/dispatchframework/javabaseimage/Server.java

```diff
@@ -0,0 +1,158 @@
+package io.dispatchframework.javabaseimage;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.function.BiFunction;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import io.undertow.Handlers;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.Undertow;
+import io.undertow.util.Headers;
+
+public class Server {
+    public static void main(String[] args) {
+        Undertow.builder()
+                .addHttpListener(8080, "0.0.0.0")
+                .setHandler(Handlers.path()
+                        .addPrefixPath("/", new ExecFunction(args[0], args[1]))
+                        .addExactPath("/healthz", new Healthz()))
+                .build()
+                .start();
+    }
+
+    public static class ExecFunction implements HttpHandler {
+        private String packageName;
+        private String className;
+        private final Gson gson;
+
+        public ExecFunction(String packageName, String className) {
+            this.packageName = packageName;
+            this.className = className;
+            this.gson = new GsonBuilder().serializeNulls().create();
+        }
+
+        public Type[] getBiFunctionTypes(Class<?> functionClass) {
+            Type[] genericInterfaces = functionClass.getGenericInterfaces();
+            Type[] genericTypes = null;
+
+            for (Type genericInterface : genericInterfaces) {
+                if (genericInterface instanceof ParameterizedType) {
+                    ParameterizedType parameterizedType = (ParameterizedType) genericInterface;
+                    if (parameterizedType.getRawType().equals(BiFunction.class)) {
+                        genericTypes = parameterizedType.getActualTypeArguments();
+                        break;
+                    }
+                }
+            }
+
+            return genericTypes;
+        }
+
+        public Object applyFunction(String message, BiFunction f, Type[] biFunctionTypes) {
+            JsonObject rootObj = new JsonParser().parse(message).getAsJsonObject();
+            String context = rootObj.get("context").toString();
+            String payload = rootObj.get("payload").toString();
+
+            if (biFunctionTypes != null && biFunctionTypes.length >= 2) {
```

Comment:
There's no point in checking this: by the time the function gets invoked, `biFunctionTypes` should already be ready to use and made sure it has exactly 3 elements (2 args' and return types). It should be a final field of `ExecFunction`.